### ISSUE TITLE
Terramaster NAS F2-422 Intel QSV does not work

### ISSUE THREAD

**Issue by blacknwhitebeard** [jlesage/docker-handbrake#252]
I am running the Handbrake docker container in my F2-422 Terramaster NAS.
This is my docker-compose configuration file:
`version: '3'
services:
  handbrake:
    image: jlesage/handbrake
    container_name: handbrake
    ports:
      - "5800:5800"
    volumes:
      - "/appdata/handbrake:/config:rw"
      - "/public:/storage:ro"
      - "/public/HandBrake/watch:/watch:rw"
      - "/public/HandBrake/output:/output:rw"
    devices:
      - "/dev/dri:/dev/dri:rw"
    environment:
      - USER_ID=0
      - GROUP_ID=0
      - LANG=it_IT
      - TZ=Europe/Rome
      - KEEP_APP_RUNNING=1
      - CONTAINER_DEBUG=1
      - DARK_MODE=1
      - VNC_PASSWORD=mypassword
    restart: unless-stopped`

Every time I try to encode there is an error.
Attached there's the activity log.
[Activity.log](https://github.com/jlesage/docker-handbrake/files/10863577/Activity.log)

I cannot understand what is the problem.
I followed the every step of the installation on the Github page.

I will appreciate every help I can get.
Thank you.

**Comment by jlesage** [comment]
Th problem you have is only with a QSV encoder ?  I guess a non-QSV encoder is working fine ?

**Comment by blacknwhitebeard** [comment]
Yes. If I choose a non-QSV encoder the process works fine and utilizes only the NAS CPU.

**Comment by blacknwhitebeard** [comment]
Hello. Just an update from my side.

Now I can't even successfully deploy the container (same compose file as in the first post).
In the logs I always see this error:
```
[cont-init   ] 10-xdg-runtime-dir.sh: executing...

[cont-init   ] 10-xdg-runtime-dir.sh: mkdir: cannot create directory ‘’: No such file or directory

[cont-init   ] 10-xdg-runtime-dir.sh: terminated with error 1.
```

Here's the complete log.
[_handbrake_logs.txt](https://github.com/jlesage/docker-handbrake/files/10979354/_handbrake_logs.txt)

**Comment by blacknwhitebeard** [comment]
Hello. Another update.

The problem I had in the previous post was only an issue with user permission: the container works fine if I use
```
    environment:
      - USER_ID=0    #my root user PUID
      - GROUP_ID=3   #my admin user group PGID
```


About the main issue I wrote about in the first post, I did some research online and I found out some information in the issues session of the Handbrake GitHub repository.
https://github.com/HandBrake/HandBrake/issues/4941#issuecomment-1435665018

If I set the "lowpower=0" parameter in the video tab as suggested in the comment, the QSV encoder works fine.

Unfortunately this brought another, stranger problem: sometime during the encoding process the entire NAS just freezes and become unusable and unreachable. I need to restart it manually because I can't even reach it through SSH.
I tried looking for some clues in the container and in the encoding process log files but found nothing. Here they are.
[_handbrake_logs.txt](https://github.com/jlesage/docker-handbrake/files/11048935/_handbrake_logs.txt)
[video_test_file 2023-03-22 16-16-24.log](https://github.com/jlesage/docker-handbrake/files/11048937/video_test_file.2023-03-22.16-16-24.log)

I can't figure out why the whole NAS crashes because of an encoding process.
I only tried with QSV H.265 with "lowpower=0" attrubute. Maybe it will be better with H.264? I don't think so but I will give it a try.

Help/suggestions are welcome.

**Comment by tschaerni** [comment]
> I can't figure out why the whole NAS crashes because of an encoding process. I only tried with QSV H.265 with "lowpower=0" attrubute. Maybe it will be better with H.264? I don't think so but I will give it a try.
> 
> Help/suggestions are welcome.

Could be because you haven't enabled the GuC/HuC Firmware, which is necessary for Casper Lake CPUs.
